vlasovsolver/cpu_trans_map_amr.cpp

```diff
@@ -1097,103 +1147,94 @@ bool trans_map_1d_amr(const dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>&
    for(const auto blockGID:  unionOfBlocksSet) {
       unionOfBlocks.push_back(blockGID);
    }
+   phiprof::stop("buildBlockList");
    // ****************************************************************************
    
+   // Compute spatial neighbors for target cells.
+   // For targets we need the local cells, plus a padding of 1 cell at both ends
+   phiprof::start("computeSpatialTargetCellsForPencils");
+   std::vector<SpatialCell*> targetCells(pencils.sumOfLengths + pencils.N * 2 );
+   computeSpatialTargetCellsForPencils(mpiGrid, pencils, dimension, targetCells.data());
+   phiprof::stop("computeSpatialTargetCellsForPencils");
+   
+   // Assuming 1 neighbor in the target array because of the CFL condition
+   // In fact propagating to > 1 neighbor will give an error
+   const uint nTargetNeighborsPerPencil = 1;
+   
+   phiprof::stop("setup");
+   
    int t1 = phiprof::initializeTimer("mapping");
    int t2 = phiprof::initializeTimer("store");
    
 #pragma omp parallel
    {
+      // declarations for variables needed by the threads
+      std::vector<Realf, aligned_allocator<Realf, 64>> targetBlockData((pencils.sumOfLengths + 2 * pencils.N) * WID3);
+      std::vector<std::vector<SpatialCell*>> pencilSourceCells;
+      
+      // Allocate aligned vectors which are needed once per pencil to avoid reallocating once per block loop + pencil loop iteration
+      std::vector<std::vector<Vec, aligned_allocator<Vec,64>>> pencilTargetValues;
+      std::vector<std::vector<Vec, aligned_allocator<Vec,64>>> pencilSourceVecData;
+      std::vector<std::vector<Vec, aligned_allocator<Vec,64>>> pencildz;
+      
+      for(uint pencili = 0; pencili < pencils.N; ++pencili) {
```

Comment:
Do we allocate too much memory here, as this loop isn't threaded? I.e. enough memory to cover all threads for each thread separately? I guess not as each thread below is for a separate block, and these allocators are for one block each?

Reply:
We are in the parallel region so each thread allocates its own of the three guys above.

Reply:
And they only need one each.